sky/utils/common_utils.py

```diff
@@ -128,11 +166,13 @@ def make_cluster_name_on_cloud(cluster_name: str,
       2. Append the hash of the cluster name
 
     Args:
-        cluster_name: The cluster name to be truncated and hashed.
+        local_cluster_name: The cluster name to be truncated and hashed.
         max_length: The maximum length of the cluster name. If None, no
             truncation is performed.
         add_user_hash: Whether to append user hash to the cluster name.
     """
+    cluster_name = _adjust_cluster_name(local_cluster_name)
```

Comment:
It seems we only adjust the cluster name that is used on the cloud, which means our internal code will still consider `sky launch -c lora` and `sky launch -c LoRA` as different clusters, i.e., it will two different entries in our state table?
This is unexpected, as our code will use the cluster name as the key to check if the user is trying to launch the same cluster.

Reply:
Good catch! If I understand correctly, the problem is: `sky launch -c lora` and `sky launch -c LoRA` would lead to the same cloud cluster name with the current implementation of `_adjust_cluster_name`, so there'd be two rows in the state table unexpectedly referring to one cloud cluster. If this understanding is correct, we could solve this problem by making sure `_adjust_cluster_name` creates a unique cloud cluster name whenever the `local_cluster_name` param is unique (e.g. by including part of hash of `local_cluster_name`)

Reply:
I am wondering if we can make the cluster name case insensitive locally, i.e. we don't separate the `sky launch -c lora` and `sky launch -c LoRA` in our database either?

Reply:
Fixed!